modules/ml/src/main/java/org/apache/ignite/ml/naivebayes/gaussian/GaussianNaiveBayesTrainer.java

```diff
@@ -0,0 +1,297 @@
+
+                    if (!res.featureSum.containsKey(label)) {
+                        toMeans = new double[features.size()];
+                        Arrays.fill(toMeans, 0.);
+                        res.featureSum.put(label, toMeans);
+                    }
+
+                    if (!res.featureCount.containsKey(label)) {
+                        res.featureCount.put(label, 0);
+                    }
+                    res.featureCount.put(label, res.featureCount.get(label) + 1);
+
+                    toMeans = res.featureSum.get(label);
+                    for (int j = 0; j < features.size(); j++) {
+                        toMeans[j] += features.get(j);
+                    }
+                }
+                return res;
+            }, (a, b) -> {
+                if (a == null)
+                    return b == null ? new MeanHelper() : b;
+                if (b == null)
+                    return a;
+                return a.merge(b);
+            });
+    }
+
+    /**
+     * Calculates squared diffs between each value of the feature and the mean.
+     *
+     * @param meansHolder means of all featured for all labels.
+     */
+    private VarianceHelper computeVariance(Dataset<EmptyContext, LabeledVectorSet<Double, LabeledVector>> dataset,
+        Map<Double, double[]> meansHolder) {
+        return dataset.compute(
+            data -> {
+                VarianceHelper res = new VarianceHelper(meansHolder);
+                for (int i = 0; i < data.rowSize(); i++) {
+                    LabeledVector row = data.getRow(i);
+                    Vector features = row.features();
+                    Double label = (Double)row.label();
+
+                    double[] m = res.meansHolder.get(label);
+
+                    double[] toVars;
+
+                    if (!res.featureDiff.containsKey(label)) {
+                        toVars = new double[features.size()];
+                        Arrays.fill(toVars, 0.);
+                        res.featureDiff.put(label, toVars);
+                    }
+
+                    toVars = res.featureDiff.get(label);
+                    for (int j = 0; j < features.size(); j++) {
+                        toVars[j] += (m[j] - features.get(j)) * (m[j] - features.get(j));
+                    }
+                }
+
+                return res;
+            }, (a, b) -> {
+                if (a == null)
+                    return b == null ? new VarianceHelper(meansHolder) : b;
+                if (b == null)
+                    return a;
+                return a.merge(b);
+            });
+    }
+
+    /** Service class is used to calculate meanses. */
+    private static class MeanHelper implements Serializable {
+        /** Serial version uid. */
+        private static final long serialVersionUID = 1L;
+        /** Sum of all values for all features for each label */
+        ConcurrentHashMap<Double, double[]> featureSum = new ConcurrentHashMap<>();
```

Comment:
I use concurrent maps because It's a copy-pasted approach from [ANN CentrodStat](https://github.com/apache/ignite/blob/master/modules/ml/src/main/java/org/apache/ignite/ml/knn/ann/ANNClassificationTrainer.java#L350). But your arguments sound reasonable. I replaced concurrent maps to hash maps